code/game/turfs/unsimulated/planetary.dm

```diff
@@ -30,6 +30,13 @@
 	if(new_temperature == temperature)
 		return
 	temperature = new_temperature
+
+	// Atmospheric pressure is *mostly* invariant to weather, so cancel out standard P~=T
+	oxygen = initial(oxygen) * initial(temperature) / new_temperature
+	nitrogen = initial(oxygen) * initial(temperature) / new_temperature
+	carbon_dioxide = initial(oxygen) * initial(temperature) / new_temperature
+	phoron = initial(oxygen) * initial(temperature) / new_temperature
```

Comment:
```suggestion
	nitrogen = initial(nitrogen) * initial(temperature) / new_temperature
	carbon_dioxide = initial(carbon_dioxide) * initial(temperature) / new_temperature
	phoron = initial(phoron) * initial(temperature) / new_temperature
```

Reply:
that could've gone badly :rofl:
![https://i.imgur.com/CFMrasK.png](https://i.imgur.com/CFMrasK.png)

Reply:
:P